zone/tradeskills.cpp

```diff
@@ -1866,37 +1866,53 @@ bool ZoneDatabase::DisableRecipe(uint32 recipe_id)
 bool Client::CheckTradeskillLoreConflict(int32 recipe_id)
 {
 	auto recipe_entries = TradeskillRecipeEntriesRepository::GetWhere(
-			content_db,
-			fmt::format(
-					"recipe_id = {} ORDER BY componentcount DESC",
-					recipe_id
-			)
+		content_db,
+		fmt::format(
+			"recipe_id = {} ORDER BY componentcount DESC",
+			recipe_id
+		)
 	);
 	if (recipe_entries.empty()) {
 		return false;
 	}
 
-	for (auto& f : recipe_entries) {
-		for (auto &e: recipe_entries) {
-			if (f.componentcount > 0 && e.item_id == f.item_id && e.componentcount == 0) {
-				e.item_id = 0;
+	const EQ::ItemData* fi = nullptr;
+	const EQ::ItemData* ei = nullptr;
+	for (auto &f : recipe_entries) {
```

Comment:
In this case, what is `f` and what is `e` ? 

It would be good to have some comments above this describing what is happening

Reply:
okay yeah good idea, changed the names to be more descriptive... and added commenting, hopefully it clears it up?